Source/WebCore/Modules/webxr/XRGPUSubImage.cpp

```diff
@@ -82,7 +81,7 @@ ExceptionOr<Ref<GPUTexture>> XRGPUSubImage::colorTexture()
     if (!texture)
         return Exception { ExceptionCode::InvalidStateError };
 
-    return GPUTexture::create(texture.releaseNonNull(), textureDescriptor(GPUTextureFormat::Bgra8unormSRGB), m_device);
+    return GPUTexture::create(texture.releaseNonNull(), textureDescriptor(GPUTextureFormat::Bgra8unorm, m_width, m_height), m_device);
```

Comment:
This is `Bgra8unorm` value is hard-coded in a lot of places. Should it be queried from some configuration state instead?

Reply:
Yes, that is correct. I plan to do so when adding HDR support https://bugs.webkit.org/show_bug.cgi?id=296675

There is also the issue that bgra8-srgb isn't a valid preferred canvas format in WebGPU but I think that is the default on visionPro. Rather, bgra8 is returned to JS and the requester asks for a canvas format of bgra8 with a view format of bgra8-srgb. I need to think how to make that work correctly as well